rust/src/database/session.rs

```diff
@@ -142,6 +161,21 @@ impl Session {
         session_info.on_close_register_sink.send(Box::new(callback)).ok();
     }
 
+    /// Registers a callback function which will be executed when this session is reopened.
```

Comment:
Extend comment so users can self-drive more easily (eg. obvious next question: when is a session repoened)